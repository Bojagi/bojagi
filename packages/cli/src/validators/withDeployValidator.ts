import { emojiMesage } from '../utils/terminalUtils';

export class ValidationError extends Error {
  hideStackTrace: boolean = true;
}

export default function withDeployValidator(action) {
  return args => {
    if (!args.commit) {
      renderError();
      throw new ValidationError(
        'Please add a commit in your arguments (e.g. --commit f9xa5a7)'
      );
    }

    if (!process.env.BOJAGI_SECRET) {
      renderError();
      throw new ValidationError(
        'Please define an environment variable called "BOJAGI_SECRET" that holds the repository secret'
      );
    }

    return action(args);
  };
}

function renderError() {
  emojiMesage(
    '‚ùå',
    'Your Inputs are not valid, please check the message below üëá',
    false,
    false
  );
}
