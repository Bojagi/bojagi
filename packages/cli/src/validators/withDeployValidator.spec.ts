import withDeployValidator from './withDeployValidator';

let wrappedAction;
let action;
let tempProcess;

function storeProcess() {
  tempProcess = { ...process.env };
}

function restoreProcess() {
  process.env = tempProcess;
}

beforeEach(storeProcess);

beforeEach(() => {
  action = jest.fn();
  wrappedAction = withDeployValidator(action);
  process.env.BOJAGI_SECRET = 'some_secret';
});

afterEach(restoreProcess);

test('fail because of missing commit', () => {
  expect(() =>
    wrappedAction({
      commit: undefined
    })
  ).toThrowErrorMatchingSnapshot();
});

test('fail because of missing secret', () => {
  process.env.BOJAGI_SECRET = undefined;
  expect(() =>
    wrappedAction({
      commit: 'abc123'
    })
  ).toThrowErrorMatchingSnapshot();
});

test('succeed and call action with args', () => {
  action.mockReturnValueOnce('action_result');
  const result = wrappedAction({
    commit: 'abc123'
  });
  expect(result).toBe('action_result');
});
