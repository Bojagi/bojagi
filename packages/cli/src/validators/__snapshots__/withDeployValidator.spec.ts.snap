// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fail because of missing commit 1`] = `"Please add a commit in your arguments (e.g. --commit f9xa5a7)"`;

exports[`fail because of missing secret 1`] = `"Please define an environment variable called \\"BOJAGI_SECRET\\" that holds the repository secret"`;
